setup.py

```diff
@@ -15,7 +15,7 @@
     "numpy>=1.18",
     "scipy>=1.5.2",
     "tqdm>=4.56.2",
-    "multipledispatch>=0.6.0",
+    "plum-dispatch~=0.2.5",
```

Comment:
I definitely approve this, and we should do it for all dependencies, but then we should setup dependabot to keep version constraints up to date...

Reply:
(not for this PR, but I'm just raising the issue)